src/main/java/org/jabref/model/search/rules/ContainBasedSearchRule.java

```diff
@@ -1,27 +1,43 @@
 package org.jabref.model.search.rules;
 
+import java.io.IOException;
 import java.util.Iterator;
 import java.util.List;
 import java.util.Locale;
+import java.util.Vector;
 
+import org.jabref.logic.pdf.search.retrieval.PdfSearcher;
 import org.jabref.model.entry.BibEntry;
 import org.jabref.model.entry.field.Field;
+import org.jabref.model.pdf.search.PdfSearchResults;
+import org.jabref.model.pdf.search.SearchResult;
 
 /**
  * Search rule for contain-based search.
  */
 public class ContainBasedSearchRule implements SearchRule {
 
     private final boolean caseSensitive;
+    private final boolean fulltext;
 
-    public ContainBasedSearchRule(boolean caseSensitive) {
+    private String lastQuery;
+    private List<SearchResult> lastSearchResults;
+
+    public ContainBasedSearchRule(boolean caseSensitive, boolean fulltext) {
```

Comment:
I think you could use an enum here for all modes and combine them? Like an EnumSet, then you can get rid of the booleans https://eddmann.com/posts/using-bit-flags-and-enumsets-in-java/

Reply:
The values for caseSensitive and fulltext mainly comes from the state of the respective toggle button (a boolean). If we change to an EnumSet, you would have to construct the EnumSet from the booleans. I don't think that would make the code any better.

Reply:
Depends. Readability is the leading concern...
See SearchQuery etc. above and below

Reply:
TBH, I don't think enum-sets are easier to read than booleans. If I look at a signature and see it expects an EnumSet, I need to check that set to understand what the options are. There is no need to do that with (named) booleans.
For my argument that I would need to convert them from the toggle-buttons, we could do that like described [here](https://stackoverflow.com/a/35851125/5401297).

Reply:
I converted to EnumSets in 44c975f6a.
I stopped at SearchPreferences not to mess with peoples preferences.